src/test/java/dk/sdu/se_f22/sortingmodule/range/rangepublic/LongFilterTest.java

```diff
@@ -75,17 +84,317 @@ void useFilter() {
-        void filteringAnEmptyListOfResults() {
-            LongFilter internalFilter = new LongFilter(0, "test name", "test description", "price", 0, 1000);
-            internalFilter.setUserMax(1000);
-            internalFilter.setUserMin(0);
-            Collection<Product> emptyResults = new ArrayList<>();
+        @ParameterizedTest(name = "setUserMinAndMax={1} - inputList={2}")
+        @MethodSource("filteringAnEmptyListOfResultsArgument")
+        void filteringAnEmptyListOfResults(boolean setUserMinAndMax, Collection<Product> inputList) throws IllegalMinMaxException {
+            LongFilter internalFilter = new LongFilter(0, "test name", "test description", "ean", 0, 1000);
+            if(setUserMinAndMax){
+                internalFilter.setUserMax(100);
+                internalFilter.setUserMin(10);
+            }
+
+            // preparing the expected result
+            // Note: we create a new variable that doesn't reference inputlist since it might be modified during the action
+            Collection<Product> expectedResult;
+
+            if(inputList == null){
+                expectedResult = null;
+            }else {
+                expectedResult = new ArrayList<>();
+            }
+
+            Collection<Product> filteredResults = internalFilter.useFilter(inputList);
+
+            assertEquals(expectedResult, filteredResults);
+        }
+
+        static Stream<Arguments> filteringAnEmptyListOfResultsArgument() {
+            List<Arguments> out = new ArrayList<>();
+
+            // create the combination of desired inputs
+            // The order of the for loops determine the order of test cases.
+            // Currently it will switch productAttribute first, then switch between using user settings or not. Then it will switch to the null list
+            for (boolean nullInsteadOfList : new boolean[]{false, true}) {
+                for (boolean setUserMinAndMax : new boolean[]{false, true}) {
+                    if(nullInsteadOfList){
+                        out.add(arguments(setUserMinAndMax, null));
+                    }else {
+                        out.add(arguments(setUserMinAndMax, new ArrayList<Product>()));
+                    }
+                }
+            }
+
+            return out.stream();
+        }
+    }
+
+    @Nested
+    @DisplayName("Set userMin and userMax")
+    class SetUserMinAndMax {
+
+        static Stream<LongFilter> provideLongFiltersProvider (){
+            return Stream.of(
+                    new LongFilter("Long filter 1", "Desc long filter 1", "price", 0, 1000),
+                    new LongFilter("Long filter 2", "Desc long filter 2", "price", 0, 30)
+            );
+        }
+
+        @Nested
+        @DisplayName("Set valid userMin and userMax")
+        class SetValidUserMinAndUserMax {
+
+            static Stream<LongFilter> provideLongFilters(){
+                return provideLongFiltersProvider();
```

Comment:
Why do this? Why not directly reference "provideLongFiltersProvider" instead of "provideLongFilters" in the methodsScource for the tests?

Reply:
It this particular nested class when trying to reference the provider directly Junit cannot find the original provider and thus fails the test with a red mark.
Should this be added as a comment?